pandas/tests/extension/test_arrow.py

```diff
@@ -1068,6 +1068,14 @@ def test_comp_masked_numpy(self, masked_dtype, comparison_op):
         expected = pd.Series(exp, dtype=ArrowDtype(pa.bool_()))
         tm.assert_series_equal(result, expected)
 
+    def test_assign_column_in_dataframe(self, data):
```

Comment:
Does this work for Series assignment as well?

Reply:
I'm not sure to understand, but it works when we assign a Series (with a `dtype` equals to `ArrowDtype`) to a DataFrame column (I completed the test to take this case into account).

Reply:
Is that right ?

Reply:
Sorry I said assignment but I meant construction:

```python
>>> ser = pd.Series(pa.array([1, 2, 3], type=pa.int16()))
>>> ser.dtype
dtype('int16')
```

I expect that should return the pyarrow int16 type but it does not - does this PR fix that too?

Reply:
(I added a test test_create_series_dtype to check it)